config/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/client/OidcBackChannelLogoutHandler.java

```diff
@@ -137,6 +145,9 @@ String computeLogoutEndpoint(HttpServletRequest request) {
 		int port = uriComponents.getPort();
```

Comment:
shouldn't the `server.port` property be used here instead of the original request port?

Reply:
I've just caught up in the discussion in https://github.com/spring-projects/spring-security/issues/14553 and am now better versed in the complexities around using `localhost`. Instead, I think a more reasonable default is `{baseUrl}/logout/connect/back-channel/{registrationId}`.

The reason it is still necessary to specify the path is that it needs to remain compatible with the previous approach, which will still be available for a time. Also, this is very similar to other OAuth 2.0 URI default configuration values.

Reply:
Honestly, the current code is reasonable as, thanks to the URI configuration, I could get back-channel logout work with a SPA using cookie-based protection against CSRF.

We are talking about what could make a better default, and yes, using `{baseUrl}` is probably one.